v2/yarpcjson/codec_test.go

```diff
@@ -0,0 +1,78 @@
+package yarpcjson
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.uber.org/yarpc/v2"
+)
+
+func TestNewCodec(t *testing.T) {
+	tests := []struct {
+		Name string
+		Func interface{}
+	}{
+		{
+			"foo",
+			func(context.Context, *struct{}) (*struct{}, error) {
+				return nil, nil
+			},
+		},
+		{
+			"bar",
+			func(context.Context, map[string]interface{}) (*struct{}, error) {
+				return nil, nil
+			},
+		},
+		{
+			"baz",
+			func(context.Context, map[string]interface{}) (map[string]interface{}, error) {
+				return nil, nil
+			},
+		},
+		{
+			"qux",
+			func(context.Context, interface{}) (map[string]interface{}, error) {
+				return nil, nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		newCodec(tt.Name, tt.Func)
```

Comment:
Let’s validate the behavior with some assertions here.

Reply:
Yea, maybe assert that the returned `jsonCodec.reader` is what we would expect?

Reply:
Unresolving this one, is there nothing we can validate here? Looks like we could add another field to our test struct that holds the expected reader type like alex suggested